src/pages/index.js

```diff
@@ -6,13 +6,44 @@ import { PopupWithForm } from "../components/PopupWithForm.js";
 import { PopupWithImage } from "../components/PopupWithImage.js"; 
 import { Section } from "../components/Section.js";
 import { UserInfo } from "../components/UserInfo.js";
+import { Api } from "../components/Api.js";
 import { 
-    initialCards, 
     options, 
-    initialName, 
-    initialDesc 
+    domain,
+    token,
 } from "../utils/constants.js";
 import "./index.css";
+import { Popup } from "../components/Popup.js";
+
+// #endregion
+
+// #region Profile setup
+
+const api = new Api({
+    baseUrl: domain,
+    headers: {
+      authorization: token,
+      "Content-Type": "application/json"
+    }
+  });
+
+const userInfo = new UserInfo(".profile__name", ".profile__description");
+
+setProfileInfo(".profile__image");
+
+function setProfileInfo(avatarSelector)
+{
+    api.getProfileInfo()
+        .then((res) => {
+            const userData = {
+                name: res.name,
+                description: res.about,
+            }
+            userInfo.setUserInfo(userData);
+            document.querySelector(avatarSelector)
+                .setAttribute("src", res.avatar);
+        });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**